os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -200,9 +214,20 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	 */
 	loadinfo->rosize += loadinfo->datasize - loadinfo->binp->bsssize;
 
+#ifdef CONFIG_ARMV7M_MPU
+	/* ARMV7M requires MPU region size to be a power of two */
 	loadinfo->textsize = 1 << mpu_log2regionceil(0, loadinfo->textsize);
 	loadinfo->rosize = 1 << mpu_log2regionceil(0, loadinfo->rosize);
 	datamemsize = 1 << mpu_log2regionceil(0, datamemsize);
+#elif CONFIG_ARMV8M_MPU
+	/* ARMV8M requires MPU region size to be aligned to 32 bytes */
+	loadinfo->textsize = MPU_ALIGN_UP(loadinfo->textsize);
```

Comment:
As above my comment, if we don't need to concatenate the regions, then we can just call memalign with 32bytes in here.

Reply:
Done.